torch/_inductor/lowering.py

```diff
@@ -186,18 +186,19 @@ def transform_args(args, broadcast, type_promotion_kind, convert_input_to_bool):
             dtype = get_promoted_dtype(
                 *promoting_args, type_promotion_kind=type_promotion_kind
             )
+
         # sometimes args are an immutable list so we can't mutate them
-        new_args = []
-        for i in range(len(args)):
-            if i in indices:
-                new_args.append(to_dtype(args[i], dtype))
-            elif isinstance(args[i], ir.Constant):
-                new_args.append(
-                    ir.Constant(args[i].value, dtype, args[indices[0]].get_device())
-                )
+        def promote(arg):
+            if isinstance(arg, TensorBox):
+                if arg.get_dtype() != dtype:
+                    return to_dtype(arg, dtype)
+                return arg
```

Comment:
This is the only functional change here. It fixes an infinite recursion issue created by `to_dtype` calling `clone`, which is wrapped with type promotion that itself calls `to_dtype`. This only happens for no-op dtype conversions, where it was wasteful to clone anyway so better to not call `to_dtype` when the dtypes already match here.